addons/web/static/src/boot.js

```diff
@@ -2,450 +2,139 @@
-    globalThis.addEventListener("load", function logWhenLoaded() {
-        const len = jobPromises.length;
-        Promise.all(jobPromises).then(function () {
-            if (len === jobPromises.length) {
-                odoo.log();
+        startModules() {
+            let job;
+            while ((job = this.findJob())) {
+                this.startModule(job);
+            }
+            if (this.failed.size || this.pending.size) {
+                this.status = "failed";
+                this.reportErrors();
             } else {
-                logWhenLoaded();
+                this.status = "idle";
             }
-        });
-    });
-
-    /**
-     * Visit the list of jobs, and return the first found cycle, if any
-     *
-     * @param {any[]} jobs
-     * @returns {null | string} either a string describing a cycle, or null
-     */
-    function findCycle(jobs) {
-        // build dependency graph
-        const dependencyGraph = new Map();
-        for (const job of jobs) {
-            dependencyGraph.set(job.name, job.dependencies);
         }
 
-        // helpers
-        function visitJobs(jobs, visited = new Set()) {
-            for (const job of jobs) {
-                const result = visitJob(job, visited);
-                if (result) {
-                    return result;
-                }
+        startModule(name) {
+            const mods = this.modules;
+            const mod = mods.get(name);
+            const require = (name) => mods.get(name).value;
+            this.pending.delete(name);
+            try {
+                mod.value = mod.factory(require);
+                // if (mod.value instanceof Promise) {
+                //     debugger
+                // }
+                this.ready.add(name);
+            } catch (error) {
+                this.failed.add(name);
+                console.error(`Error while loading "${name}":\n`, error);
             }
-            return null;
         }
 
-        function visitJob(job, visited) {
-            if (visited.has(job)) {
-                const jobs = Array.from(visited).concat([job]);
-                const index = jobs.indexOf(job);
-                return jobs
-                    .slice(index)
-                    .map((j) => `"${j}"`)
-                    .join(" => ");
-            }
-            const deps = dependencyGraph.get(job);
-            return deps ? visitJobs(deps, new Set(visited).add(job)) : null;
-        }
+        async reportErrors() {
+            const failed = this.failed;
+            const unloaded = this.pending;
 
-        // visit each root to find cycles
-        return visitJobs(jobs.map((j) => j.name));
-    }
 
-    function displayModuleErrors({ failed, missing, unloaded, cycle }) {
-        const list = (heading, arr) => {
-            const frag = document.createDocumentFragment();
-            if (!arr || !arr.length) {
+            await owl.whenReady(); // we need the DOM to be ready
```

Comment:
I don't think this should depend on owl.

Reply:
indeed.